src/main/java/io/spokestack/spokestack/nlu/NLURequest.java

```diff
@@ -0,0 +1,25 @@
+package io.spokestack.spokestack.nlu;
+
+/**
+ * A data class that represents information sent to an NLU service for
+ * classification.
+ */
+public final class NLURequest {
```

Comment:
I don't think you need a class that encapsulates a string - just pass the utterance string to the nlu

Reply:
I originally was going to do it that way, but it's wrapped like this specifically to allow extension to third-party NLUs like Dialogflow that let you pass a context object along with your utterance. If I had it just as a String, that extension would become either a breaking API change or an overload. Would one of those be preferable to making the Spokestack version a little more inconvenient, like this does?

Reply:
I think if you have a generic context object, then your request function just needs two parameters: utterance and context
I can't think of anything else to pass that would require a whole class - if it turns out that you need more than three, it would make sense to refactor, but I don't think we're there yet

Reply:
I was going off [this](https://dialogflow.com/docs/reference/agent/query), but you're right that all that can be handled in a generic context object and extracted by the NLU implementation. I'm fine with that.